File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/service/ServiceProvider.java
package com.trtc.uikit.livekit.service;

import com.trtc.uikit.livekit.service.impl.LiveServiceImpl;

public class ServiceProvider {

    private ServiceProvider() {
    }

    private static class ServiceHolder {
        private static final ServiceProvider instance = new ServiceProvider();
    }

    public static ServiceProvider getInstance() {
        return ServiceHolder.instance;
    }

    public ILiveService getLiveService() {
        return new LiveServiceImpl();
    }
}

File: Flutter/lib/service/service_provider.dart
import 'package:tencent_live_uikit/service/impl/live_service_impl.dart';
import 'package:tencent_live_uikit/service/live_service.dart';

class ServiceProvider {
  ServiceProvider._();

  static final ServiceProvider instance = ServiceProvider._();

  ILiveService getLiveService() {
    return LiveServiceImpl();
  }
}
